src/main/java/org/betonquest/betonquest/events/VariableEvent.java

```diff
@@ -29,8 +30,10 @@ public VariableEvent(final Instruction instruction) throws InstructionParseExcep
         objectiveID = instruction.getObjective();
         key = instruction.next();
         keyVariables = BetonQuest.resolveVariables(key);
-        value = instruction.next();
-        valueVariables = BetonQuest.resolveVariables(value);
+        final List<String> args = new ArrayList<>(instruction.size());
```

Comment:
You should be able to use the new required arguments getter here (and maybe void the first argument?)